src/docs/arch-hs.dox

```diff
@@ -1,8 +1,123 @@
+
+History store cursor users must use one of the flags if the caller thread is running in a lower
+isolation level and doesn't hold a valid snapshot.
+
+@section arch_hs_reconciliation History store and reconciliation
+When a dirty page is reconciled on a user file b-tree, the update chain is examined and the latest
+committed update is chosen as the on-disk value. All older updates are added to the history store
+table assuming they are not yet obsolete. Additionally any out of order timestamps will
+be corrected.
+
+When a modified history store table page is processed by reconciliation, certain records on the page
+are removed when creating a new page image if the tombstones on those records have become globally
+visible. This ensures that WiredTiger is only keeping the relevant historical data required to
+serve the oldest reader in the system or as dictated by the oldest timestamp set by the application.
+Once all records on a page are obsolete, the page itself can be removed to reduce the size of
+the history store table (see @ref arch-transaction for more details about oldest timestamp).
+
+@section arch_hs_read Searching for older versions of a key in History Store
+When looking for an update visible to the current transaction, WiredTiger first searches the update
+chain for any visible updates. If there is no visible update in the chain, WiredTiger then
+inspects the on-disk version of the key. If that version is not visible to the transaction
+as well, WiredTiger searches the history store table for a visible update for the key. When
+searching for update that satisfies the read timestamp constraint, WiredTiger starts with the newest
+update of the key in the history store table and then iterates through the older updates until there
+are no updates left to process. Note that although there can be multiple updates in the
+history store for a key and read timestamp combination, each update would have different
+visibility based on transaction and timestamp based visibility rules.
+
+@section arch_hs_rts History tore and rollback-to-stable
+Rollback-to-stable searches the history store table for valid updates that are stable according to
+the supplied stable timestamp and replaces the on-disk version with an update from the history store
+table that fulfills the criteria. Rollback-to-stable also removes updates from the
+history store table that are not stable according to the stable timestamp and transaction snapshot
+being used for rollback-to-stable operation.
+
+@section arch_hs_prepared History store and prepared transactions
+When there is a prepared update for a key and the page is evicted, the prepared update is written
+to the on-disk page and any older updates are written to the history store table. There can be
+no update from a different transaction that is newer than the prepared update for the key and
+therefore the history store should never contain a prepared update. When a prepared update is resolved
```

Comment:
Don't saw resolved (committed) just say committed otherwise its ambiguous as to why a rollback isn't considered a resolution. To me both are valid resolutions.

Reply:
I have added text that should make it more clear.